packages/preferences-api/src/common/preference-proxy.ts

```diff
@@ -21,13 +23,35 @@ export type PreferenceEventEmitter<T> = {
     readonly ready: Promise<void>;
 };
 export type PreferenceProxy<T> = Readonly<T> & Disposable & PreferenceEventEmitter<T>;
-export function createPreferenceProxy<T extends Configuration>(preferences: PreferenceService, configuration: T): PreferenceProxy<T> {
+export function createPreferenceProxy<T extends Configuration>(preferences: PreferenceService, configuration: T, schema: PreferenceSchema): PreferenceProxy<T> {
+    const ajv = new Ajv();
     const toDispose = new DisposableCollection();
     const onPreferenceChangedEmitter = new Emitter<PreferenceChange>();
     toDispose.push(onPreferenceChangedEmitter);
     toDispose.push(preferences.onPreferenceChanged(e => {
         if (e.preferenceName in configuration) {
-            onPreferenceChangedEmitter.fire(e);
+            if (e.newValue) {
+                // Fire the pref if it's valid according to the schema
+                if (ajv.validate(schema, {
```

Comment:
The validation should also happen then someone calls `proxy[myPreference]`, here: https://github.com/theia-ide/theia/pull/365/files#diff-92178177a3db9099cb8ab0ba76cff5c6R60